### ISSUE TITLE
Head of 2016.3 - Salt-Master uses 90 seconds to restart

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#39119]
When restarting the master on head of 2016.3, it uses about 90 seconds to restart. The minion restart fine.
```
[root@li882-223 ~]# time service salt-master restart
Redirecting to /bin/systemctl restart  salt-master.service

real	1m30.464s
user	0m0.010s
sys	0m0.003s
[root@li882-223 ~]# time service salt-minion restart
Redirecting to /bin/systemctl restart  salt-minion.service

real	0m0.153s
user	0m0.007s
sys	0m0.007s
[root@li882-223 ~]# time service salt-master restart
Redirecting to /bin/systemctl restart  salt-master.service

real	1m30.628s
user	0m0.007s
sys	0m0.010s
[root@li882-223 ~]# 
```

### Steps to Reproduce Issue
`time service salt-master restart`

### Versions Report
```
Salt Version:
           Salt: 2016.3.5-64-gbcee3d1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Sep 15 2016, 22:37:39)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 4.8.6-x86_64-linode78
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

**Comment by thatch45** [comment]
Can you tell me how much data is in the master cache? Particularly the number of files?

**Comment by frogunder** [comment]
It was a clean machine with just salt installed, I tried twice with clean machines on head of 2016.3, with same result.
```
[root@li882-223 ~]# ls /var/cache/salt/master
jobs  proc  queues  roots  syndics  tokens
[root@li882-223 ~]# du -sh /var/cache/salt/master
32K	/var/cache/salt/master
[root@li882-223 ~]#
```

**Comment by frogunder** [comment]
Confirmed, this fixes it on head of 2016.3.
Closing this issue.

**Comment by PierreR** [comment]
@frogunder It is still quite slow for me with the latest `2016.11.5` (using CentOS 6.7) even after clearing the cache:
```
[ 0 ] root # salt-call saltutil.clear_cache
local:
    True
19:39:01 jenkinsmaster ( /home/alhazen )
[ 0 ] root # time service salt-minion restart
Stopping salt-minion:root:jenkinsmaster.sandbox.srv.cirb.lan daemon: OK
Starting salt-minion:root:jenkinsmaster.sandbox.srv.cirb.lan daemon: OK

real	1m13.298s
user	0m48.754s
sys	0m18.818s
```
